src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -518,33 +518,37 @@ public static int findRepeatedSubstring(byte[] bases) {
      * @return                       Number of repetitions (0 if testString is not a concatenation of n repeatUnit's
      */
     public static int findNumberOfRepetitions(byte[] repeatUnit, byte[] testString, boolean lookForward) {
-        int numRepeats = 0;
+        final int repeatLength = repeatUnit.length;
+        final int testLength = testString.length;
+        final int lengthDifference = testLength - repeatLength;
+
         if (lookForward) {
+            int numRepeats = 0;
             // look forward on the test string
-            for (int start = 0; start < testString.length; start += repeatUnit.length) {
-                int end = start + repeatUnit.length;
-                byte[] unit = Arrays.copyOfRange(testString,start, end);
-                if(Arrays.equals(unit,repeatUnit))
+            for (int start = 0; start < testLength; start += repeatLength) {
+                if(start <= lengthDifference && Utils.equalRange(testString, start, repeatUnit, 0, repeatLength)) {
                     numRepeats++;
-                else
+                } else {
                     break;
+                }
+            }
+            return numRepeats;
+        } else {
+            // look backward. For example, if repeatUnit = AT and testString = GATAT, number of repeat units is still 2
+            int numRepeats = 0;
+            // look backward on the test string
+            for (int start = lengthDifference; start >= 0; start -= repeatLength) {
+                if (start <= lengthDifference && Utils.equalRange(testString, start, repeatUnit, 0, repeatLength)) {
```

Comment:
The `start <= lengthDifference` test is redundant here, since `start` is initialized to `lengthDifference` and decremented after that.